index.js

```diff
@@ -75,7 +111,30 @@ Babel.prototype.baseDir = function() {
 };
 
 Babel.prototype.transform = function(string, options) {
-  return transpiler.transform(string, options);
+  if (transformIsParallelizable(options)) {
+    // send the job to the worker pool
+    return new Promise(function(resolve, reject) {
+      pool.exec('transform', [string, options])
+      .then(
+        function onResolved(result) {
+          resolve(result);
+        },
+        function onRejected(err) {
+          if (err.name === 'Error' && (err.message === 'Worker terminated unexpectedly' ||
```

Comment:
`err` may be undefined or null or a number you should likely add a guard. 

`typeof err === 'object' && err !== null`